### PR TITLE
docs: improved the response fields docs of recipe

### PR DESCRIPTION
**Pull request by Peeeekay**
Changelog picked up from commits here:

docs: improved the response fields docs of recipe

### CODE REVIEW

**Comment by Peeeekay** [comment]
This PR Closes - https://app.zenhub.com/workspaces/engineering-636cff9fc978ceb2aac05a1d/issues/gh/kurtosis-tech/kurtosis-private/974

File: docs/docs/reference/starlark-instructions.md [hunk 1]
```diff
@@ -358,12 +358,17 @@ post_request_recipe = PostHttpRequestRecipe(
...
+post_response = plan.request(
     recipe = post_request_recipe,
 )
 ```
 
-NOTE: You can use the power of `jq` during your extractions. For example, `jq`'s [regular expressions](https://devdocs.io/jq-regular-expressions-pcre/) can be used to manipulate the extracted strings like so:
+The instruction returns a `dict` with following key-value pair; the values are a [future reference][future-references-reference] 
+* `post_response["code"]` - returns the future reference to the `status code` of the response 
+* `post_response["body"]` - returns the future reference to the `body` of the the response
+* `post_response["extract.some-custom-field"]` - it is an optioanl field and returns the future reference to the value extracted from `body`, which is explained below.
```
Comment (leoporoli) [inline docs/docs/reference/starlark-instructions.md]: typo here: ...it is an **optional**

Comment (leoporoli) [reply docs/docs/reference/starlark-instructions.md]: it works when the response body is a JSON object, right? if it's right I believe that will be better if we add this clarification.

File: docs/docs/reference/starlark-instructions.md [hunk 2]
```diff
@@ -177,16 +177,16 @@ plan.print(result["code"])
...
 
 ```python
 exec_recipe = ExecRecipe(
     service_name = "my_service",
     command = ["echo", "Hello, world"],
 )
 
-result = exec(exec_recipe)
-assert(result["output"], "==", 0)
+result = plan.exec(exec_recipe)
```
Comment (leoporoli) [inline docs/docs/reference/starlark-instructions.md:188]: good catch!

File: docs/docs/reference/starlark-instructions.md [hunk 3]
```diff
@@ -373,10 +378,28 @@ post_request_recipe = PostHttpRequestRecipe(
...
+NOTE: In the above example, `response` also has a custom field `extract.second-element-from-list-head` and the value is `world` which is extracted from the `response[body]`.
+
+These fields can be used in conjuction with [`assert`][assert] and [`wait`][wait] instructions, like so:
+```python
+# Following the example above, response["extract.second-element-from-list-head"] is world
+response = plan.request(
+    recipe = post_request_recipe,
+)
+
+# assert if the extracted field in the response is world
```
Comment (leoporoli) [inline docs/docs/reference/starlark-instructions.md]: just to match the current style in the docs, the first character should be uppercase

File: docs/docs/reference/starlark-instructions.md [hunk 4]
```diff
@@ -373,10 +378,28 @@ post_request_recipe = PostHttpRequestRecipe(
...
+assert(response["extract.second-element-from-list-head"], "==", "world")
+
+# make a post request and check if the extracted field in the response is world
```
Comment (leoporoli) [inline docs/docs/reference/starlark-instructions.md]: just to match the current style in the docs, the first character should be uppercase

File: docs/docs/reference/starlark-instructions.md [hunk 5]
```diff
@@ -513,7 +536,8 @@ response = plan.wait(
     # MANDATORY
     recipe = recipe,
 
-    # The field of the recipe's result that will be asserted
+    # wait will use the response's field to do the asssertions. To learn more about available fields,
```
Comment (leoporoli) [inline docs/docs/reference/starlark-instructions.md]: just to match the current style in the docs, the first character should be uppercase.

File: docs/docs/reference/starlark-instructions.md [hunk 6]
```diff
@@ -555,6 +579,8 @@ in Kurtosis Starlark by default
 <!--------------- ONLY LINKS BELOW THIS POINT ---------------------->
 [set-connection]: #set_connection
 [add-service]: #add_service
+[wait]: #wait
```
Comment (leoporoli) [inline docs/docs/reference/starlark-instructions.md:582]: great adding these links

**Review by leoporoli — APPROVED** [review]

**Review by leoporoli — APPROVED** [review]

**Pull request merged**